.github/workflows/ci-new.yml

```diff
@@ -248,3 +274,14 @@ jobs:
             -shuffle=on \
             -timeout='10m' \
             ./test/integ-new/...
+      # Keys can not be deleted, we can only destroy them.
+      # We are not charged by keys that are destried.
+      # See: https://cloud.google.com/kms/docs/faq#cannot_delete.
+      - name: 'clean up kms key'
+        if: '${{ always()  && needs.deployment.steps.create_kms_key.outcome == ''success'' }}'
```

Comment:
if: '${{ always() && steps.create_kms_key.outcome == ''success'' }}'

Also, -> "Keys cannot be deleted, we can destroy key versions to avoid billing"